### ISSUE TITLE
Disk and S3 based object stores have files remaining after purging all existing histories in Galaxy

### ISSUE THREAD

**Issue by pcm32** [galaxyproject/galaxy#15292]
**Describe the bug**

After deleting all datasets and histories from the UI, the S3 (Swift) object store still keeps elements.

The object store says:
```
>>> for bucket in conn.get_all_buckets():
...    total_bytes = 0
...    name = bucket.name
...    for key in bucket:
...      total_bytes += key.size
...    total_bytes = total_bytes/1024/1024/1024
...    print(f"Bucket Name: {name} Size: {total_bytes} GB")
...
Bucket Name: galaxy-dev Size: 17.76975029055029 GB
```

The UI says:
<img width="915" alt="image" src="https://user-images.githubusercontent.com/368478/211677522-a4949075-12a0-4333-a28d-d02c11612c38.png">

<img width="1208" alt="image" src="https://user-images.githubusercontent.com/368478/211679442-0a32e066-6ec0-4e13-a366-c3be6a4de0c0.png">

The database says:
```
galaxy=> select j.user_id, COUNT(*), SUM(file_size)/1024/1024 FROM dataset d JOIN job j ON d.job_id = j.id WHERE not purged GROUP BY j.user_id;
 user_id | count |      ?column?
---------+-------+--------------------
       1 |   328 | 83480.660652160645
```

Disk based file system says:
```
galaxy@galaxy-dev2-job-0-6dc58b685f-n5vg9:/galaxy/server/database/objects$ du -h --max-depth=1 .
8.0G	./e
2.8G	./c
3.4G	./0
2.0G	./1
3.1G	./2
3.1G	./a
2.2G	./3
4.4G	./d
3.4G	./6
4.8G	./b
5.7G	./8
4.2G	./f
3.7G	./5
6.2G	./4
4.8G	./9
2.8G	./7
64G	.
```

But those ~83 GB are nowhere to be seen on the UI :-(.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 22.05
Commit: 36f80978e1b9743f413491a51f47c21ba522c6ed

**To Reproduce**
1. Add some datasets and run some tools, ideally on an instance with a single user.
2. Maybe restart galaxy a few times
3. Delete and purge all the histories in the side by side histories view.
4. Check file system, S3 object store and database

**Expected behavior**

After deletion and purging of all datasets, I would expect object store on S3 and objects directory on disk to be mostly empty, and this should be reflected on the database.

**Comment by pcm32** [comment]
Is it safe to directly delete those elements from the disk and object store if the UI doesn't see them? Thanks!

**Comment by pcm32** [comment]
I run the maintenance.sh I think and it didn't change things. But maybe you have other cleanup scripts in mind? Could you point me to the ones that you are referring to just to be sure? I can then try those. Thanks!

What is weird is that the purge normally does reduce the disk usage (I have seen it), but for some reason I have these rogue datasets there that don't seem to go away.

**Comment by pcm32** [comment]
I run at least this [one](https://github.com/galaxyproject/galaxy/blob/dev/scripts/maintenance.sh), after applying a patch from this https://github.com/galaxyproject/galaxy/pull/15222.

**Comment by bgruening** [comment]
Can you try the cleanup_dataset.py script https://galaxyproject.org/admin/config/performance/purge-histories-and-datasets/#purging-unwanted-histories-libraries-and-datasets or this here: https://github.com/galaxyproject/galaxy/blob/dev/scripts/cleanup_datasets/pgcleanup.py

**Comment by pcm32** [comment]
I suspect that the first set of calls is covered by maintenance.sh:

https://github.com/galaxyproject/galaxy/blob/cebc02c6eaab191299ef270821dcda30210f2326/scripts/maintenance.sh#L71-L76

Maybe pgcleanup.py is not covered by maintenance.sh. Will try both again, thanks!

**Comment by pcm32** [comment]
So I tried maintenance.sh again, I got some initally promising logs:

```
######### Processing dataset id: 175
Marking as deleted: HistoryDatasetAssociation id 354 (for dataset id 175)
Deleting dataset id 175
######### Processing dataset id: 9
Dataset is not deletable (shared between multiple histories/libraries, at least one is not deleted)
######### Processing dataset id: 11
Dataset is not deletable (shared between multiple histories/libraries, at least one is not deleted)
######### Processing dataset id: 14
Dataset is not deletable (shared between multiple histories/libraries, at least one is not deleted)
######### Processing dataset id: 240
Marking as deleted: HistoryDatasetAssociation id 419 (for dataset id 240)
Deleting dataset id 240
######### Processing dataset id: 241
Marking as deleted: HistoryDatasetAssociation id 420 (for dataset id 241)
Deleting dataset id 241
######### Processing dataset id: 242
Marking as deleted: HistoryDatasetAssociation id 421 (for dataset id 242)
Deleting dataset id 242
######### Processing dataset id: 510
Marking as deleted: HistoryDatasetAssociation id 831 (for dataset id 510)
Deleting dataset id 510
```

I was surprised by the `shared between multiple histories/libraries` since I got no histories of libraries with files anymore...

but then I was disappointed to see that the final purge still didn't delete datasets:

```
# 2023-01-11 17:24:10 - Handling stuff older than 5 days
Datasets will be removed from disk.

Purged 0 datasets
Freed disk space: 0
Elapsed time: 0.626493
```

and the 64 GBs on disk continue to be present :-(.

Running `pgcleanup.py` with the different options always came back with:

```
2023-01-11 17:36:36,284 INFO  _update(): Update resulted in no changes, rolling back transaction
```

would it be safe then to delete whatever is there?

**Comment by bgruening** [comment]
@pcm32 that does all not make much sense. There is a bigger problem here. I fear you really need to go into the DB. Look at gxadmin to make life a bit easier. I would debug it now in the following way:

* go to your disk and search for a dataset that should be deleted but is present
* take this dataset it or UUID and look it up in the Galaxy database
* what is the status of this dataset, deleted or purged or ok?

**Comment by bgruening** [comment]
This dataset is not `purged` in the DB. That could explain why you see it on disc.

**Comment by pcm32** [comment]
Maybe on deletion/purging, datasets of resubmitted jobs end up in some limbo? Maybe it is related to how the k8s runner is "closing" the datasets on failure as mentioned here https://github.com/galaxyproject/galaxy/issues/15118#issuecomment-1376479912?

**Comment by bgruening** [comment]
I have no clue about this runner :(

**Comment by pcm32** [comment]
> This dataset is not purged in the DB. That could explain why you see it on disc.

So this is why the scripts don't get rid of it? I either deleted this through the purge history on the side by side view or the "Delete history" on the newer UI:

<img width="275" alt="image" src="https://user-images.githubusercontent.com/368478/211884906-4f9f9c86-7168-4479-ba6e-9d61e023d370.png">

**Comment by pcm32** [comment]
OK, there are that are some that are not resubmitted, but that show the same phenotype of not being purged (in principle, the instance has no remaining datasets that can be seen through the UI), so it is not resubmission related.

**Comment by bgruening** [comment]
> So this is why the scripts don't get rid of it?

I think so yes.

Please study those scripts, especially purge_history functionality. https://github.com/galaxyproject/galaxy/tree/dev/scripts/cleanup_datasets

I recall that in your case you need to run the script to mark datasets as purged when histories are purged and then remove them from disc, I think with the `-r` switch. There should also be a GTN training about it - sorry on a phone atm.

**Comment by pcm32** [comment]
> sorry on a phone atm.

You have been super helpful, thanks man!

**Comment by pcm32** [comment]
Ok, this sequence of commands did it (probably not all needed):

```
python scripts/cleanup_datasets/cleanup_datasets.py /galaxy/server/config/galaxy.yml -3 -r -d 1
python scripts/cleanup_datasets/cleanup_datasets.py /galaxy/server/config/galaxy.yml -6 -r -d 1
python scripts/cleanup_datasets/cleanup_datasets.py /galaxy/server/config/galaxy.yml -1 -r -d 1
python scripts/cleanup_datasets/cleanup_datasets.py /galaxy/server/config/galaxy.yml -2 -r -d 1
python scripts/cleanup_datasets/cleanup_datasets.py /galaxy/server/config/galaxy.yml -3 -r -d 1
python scripts/cleanup_datasets/cleanup_datasets.py /galaxy/server/config/galaxy.yml -3 -r -d 0
...
Purged 346 datasets
Freed disk space: 84099783580
Elapsed time: 343.417824
```

thanks @bgruening

**Comment by bgruening** [comment]
Yeah! :+1:

**Comment by pcm32** [comment]
I suspect that I got here because one of the UI based methods for erasing datasets must be:

- Hiding them completely to the user
- Yet not purging them

I will try different deletion methods and see what happens.

We start with this on disk after two large workflow executions:

```
galaxy@galaxy-dev2-job-0-5f94ddb48d-k72cq:/galaxy/server/database$ df -h .
Filesystem                                                    Size  Used Avail Use% Mounted on
10.43.9.166:/export/pvc-76bf9960-344c-4ac7-8d37-2d7c7da411df  197G   87G  111G  44% /galaxy/server/database
```

In the database we see the following hidden datasets for one of those histories:

```
SELECT d.id, d.state, d.deleted, d.purged, d.file_size, hda.visible FROM dataset d JOIN history_dataset_association hda ON d.id = hda.dataset_id WHERE hda.history_id = 42 and not hda.visible;
```


```
  id  | state | deleted | purged | file_size | visible
------+-------+---------+--------+-----------+---------
 1585 | ok    | f       | f      | 147363533 | f
 1634 | ok    | f       | f      |   2098024 | f
 1586 | ok    | f       | f      | 147154443 | f
 1617 | ok    | f       | f      | 458601079 | f
 1739 | ok    | f       | f      | 608711205 | f
 1654 | ok    | f       | f      | 475101365 | f
 1619 | ok    | f       | f      | 458604244 | f
 1731 | ok    | f       | f      | 608738596 | f
 1606 | ok    | f       | f      |         2 | f
 1607 | ok    | f       | f      |         3 | f
 1608 | ok    | f       | f      |         3 | f
...
 1744 | ok    | f       | f      | 609009140 | f
 1748 | ok    | f       | f      | 609112469 | f
 1752 | ok    | f       | f      | 609371134 | f
 1754 | ok    | f       | f      | 609431279 | f
 1756 | ok    | f       | f      | 609615379 | f
(218 rows)
```


## Deletion operation 1 (delete from history)

Deleting them through the new history UI (note that the hidden icon count and database are slightly off, maybe due to collection's parent object being counted in the eye?). I don't expect this to reduce any disk usage, as it probably just delete associations to the history:

<img width="289" alt="image" src="https://user-images.githubusercontent.com/368478/212016986-852eda3a-dab1-4f2c-8b18-120e35ea492f.png">

This operation didn't change:

- The output of the above query in the database 
- The usage of the history (database icon still shows the same number)
- The number of hidden datasets (still shows 236 hidden datasets)
- The disk use

## Use storage manager to free up disk space

Using:

<img width="917" alt="image" src="https://user-images.githubusercontent.com/368478/212019257-caea16c7-9234-4a69-b0d9-5424702b1f8b.png">

and pressing on the "Review and clear" shows us the expected 218 datasets that where hidden and then deleted. Clicking on the link to select all 218 datasets (not the check box) and then the `Permanently delete 218 items`:

<img width="696" alt="image" src="https://user-images.githubusercontent.com/368478/212020199-eba49034-0950-47be-9f36-c358021fa0a9.png">

claims to clear 47.8 GBs, however:

- Disk usage didn't change (stuck on the original 87 GBs used).
- Database still shows the 218 items not deleted and not purged!
- UI still shows hidden elements without any indication of them being deleted.
- Web app logs shows no sign of the deletion operations failing as far as I can see.
- The storage dashboard claims that the user now is using ~20 GBs less of data (but this is not translated into the actual disk usage nor database somehow, or not on the queries I'm making).

So maybe the new history UI or the new disk management setup have some issues.

## Using the legacy panel to delete

I reckon that this exercise might be biased, as we had been already poking with the history and maybe galaxy is doing some deletions asynchronously? At least I don't seem to see any of that. Even 10 minutes or so after the deletions database and disk look the same.

- Choose "Delete hidden datasets"
- Choose "Purge deleted datasets"

<img width="290" alt="image" src="https://user-images.githubusercontent.com/368478/212023712-044cea45-3dcf-46b3-b635-9476f4fb65f3.png">

- Brings down disk usage from 87 GBs to 56 GBs
- Database query now shows those 218 datasets deleted and purged.

So most likely the datasets I couldn't see and where still not deleted comes from some issue in the newer history or disk space management dashboard (which are both super cool!).

**Comment by pcm32** [comment]
Interestingly, when I switched to a sister history from the execution of the same workflow on a different dataset, this one claimed already to have all hidden datasets deleted and purged:

<img width="289" alt="image" src="https://user-images.githubusercontent.com/368478/212025986-1b7e4525-0b72-4cf0-be31-65907533d6ed.png">

however, its hidden elements still showed as not deleted nor purged in the database:

```
  id  | state | deleted | purged | file_size | visible
------+-------+---------+--------+-----------+---------
 1790 | ok    | f       | f      |         3 | f
 1806 | ok    | f       | f      | 377743792 | f
 1824 | ok    | f       | f      |         3 | f
 1834 | ok    | f       | f      |         4 | f
 1840 | ok    | f       | f      |        23 | f
 1833 | ok    | f       | f      |         4 | f
...
 1908 | ok    | f       | f      | 513683519 | f
 1910 | ok    | f       | f      | 513840107 | f
 1912 | ok    | f       | f      | 513922622 | f
 1914 | ok    | f       | f      | 514081696 | f
(218 rows)
```

but the UI insists that they are gone and that the user is using far less space than it really is. Is there any intermediate elements between UI and DB currently (something on celery or rabbit perhaps?).

**Comment by davelopez** [comment]
Thank you @pcm32 for the detailed explanations, I will investigate further on a local instance to see if we can find anything suspicious. It is indeed very strange that the DB and the UI can give different information since obviously the source is the same. Let's see if we can find an explanation for that.

**Comment by pcm32** [comment]
Thanks, let me know if I can provide any more details. I'll refrain from touching that last history as much as possible in case you need more pieces of the puzzle.

**Comment by davelopez** [comment]
Ok, I think I may have found what is confusing here. If you change your DB query like this:

```diff
- SELECT d.id, d.state, d.deleted, d.purged, d.file_size, hda.visible FROM dataset d JOIN history_dataset_association hda ON d.id = hda.dataset_id WHERE hda.history_id = 42 and not hda.visible;
+ SELECT d.id, d.state, hda.deleted, hda.purged, d.file_size, hda.visible FROM dataset d JOIN history_dataset_association hda ON d.id = hda.dataset_id WHERE hda.history_id = 42 and not hda.visible;
```
You will see that the `HistoryDatasetAssociations` are correctly deleted and purged after the corresponding operations in the UI and, as I understand, the HDA (and not the Dataset) is all the UI cares about since the actual underlying Dataset (from the datasets table) might be still used by another HistoryDatasetAssociation, library, etc.

I think that is why it still appears as *deleted=false* and *purged=false* in your query and it is only after you run the `cleanup_datasets` script that it figures out "OK, time to clean and nobody else is using this dataset, so I can finally delete it from disk".

I might be wrong though, so please someone let me know if this is not how it currently works.

The other issue about the Disk usage indicator not getting properly updated can happen sometimes. You can use the "Refresh" button in the main Storage Dashboard page to force an update, and sometimes you just need to completely refresh the page after a while or manipulate one of your histories for the value to properly show up, but we are planning on enhancing this experience in a future version.

**Comment by pcm32** [comment]
Thanks Dave, that is very useful! However, I can guarantee that those datasets are not being used in any other history (this is a dev instance where I'm the only user, and these separate histories are created by workflow executions), so I'm not sure why Galaxy would delete them from the history and say that they are purged in the UI (it actually says deleted from disk), yet the files still exist.

For instance, dataset 232:

<img width="1214" alt="image" src="https://user-images.githubusercontent.com/368478/212145688-36aeba57-6652-4c9c-b56d-249c150fcb14.png">

shows both deleted and purged in the UI. In the file system:

```
galaxy@galaxy-dev2-job-0-5f94ddb48d-k72cq:/galaxy/server/database/jobs_directory$ ls -l /galaxy/server/database/objects/7/7/e/dataset_77e9a130-6171-434f-904f-716d76793d30.dat
-rw-r--r-- 1 galaxy galaxy 374688354 Jan 12 01:06 /galaxy/server/database/objects/7/7/e/dataset_77e9a130-6171-434f-904f-716d76793d30.dat
```

very much alive. In the database:

```
galaxy=> SELECT d.id, hda.history_id, d.state, d.deleted as d_deleted, d.purged as d_purged, d.file_size as d_file_size, hda.visible, hda.deleted, hda.purged FROM dataset d JOIN history_dataset_association hda ON d.id = hda.dataset_id WHERE d.uuid LIKE '77e9a1306171434f%';
  id  | history_id | state | d_deleted | d_purged | d_file_size | visible | deleted | purged
------+------------+-------+-----------+----------+-------------+---------+---------+--------
 1863 |         46 | ok    | f         | f        |   374688354 | f       | t       | t
 1863 |         46 | ok    | f         | f        |   374688354 | f       | t       | t
 1863 |         46 | ok    | f         | f        |   374688354 | f       | t       | t
(3 rows)
```

You can see that it is in a single history. Shouldn't a dataset in a single history get purged from disk when it is purged from that history? I wonder though why it comes up 3 times instead of just once...

```
galaxy=> SELECT history_id, dataset_id, create_time, tool_version, state, name FROM history_dataset_association WHERE dataset_id = 1863;
 history_id | dataset_id |        create_time         | tool_version | state |          name
------------+------------+----------------------------+--------------+-------+-------------------------
         46 |       1863 | 2023-01-12 00:36:07.567219 |              |       | tsne_perplexity_45.h5ad
         46 |       1863 | 2023-01-12 01:11:06.383202 |              |       | tsne_perplexity_45.h5ad
         46 |       1863 | 2023-01-12 01:11:10.849327 |              |       | tsne_perplexity_45.h5ad
(3 rows)
```

ok, it seems that either resubmissions or potentially succesive collections filtering could be playing a trick here? So maybe the deletion query, when checking if a datasets is in more than one history should do a unique or a group by history perhaps? It is probably falling prey of this artifact to thinking that this is in more than one history.

**Comment by pcm32** [comment]
That dataset went through a filtering and a collections merge at the least... maybe that explains the first (original), second (filtering) and third (merging with another collection) entry?

**Comment by pcm32** [comment]
However, when deleting through the older UI, for histories that are identical in structure, I don't see the remaining files in the file system.

**Comment by davelopez** [comment]
> I can guarantee that those datasets are not being used in any other history

We can guarantee it, but as far as I know, Galaxy can't, since it doesn't have a reference count for each dataset. I believe is when we run this cleanup script when this is actually checked. But again I might be wrong.

> so I'm not sure why Galaxy would delete them from the history and say that they are purged in the UI (it actually says deleted from disk), yet the files still exist.

I understand that this may be confusing since you are both user and admin, but the UI is "the user view of Galaxy", so for the user, the dataset is deleted, and purged (which just means this won't be taken into account for the disk usage and cannot be undeleted anymore), however, from the point of view of the admin, the file might still be on disk until the cleaning process kicks in.

> I wonder though why it comes up 3 times instead of just once...

I will assume you are seeing 3 HDAs pointing to the same Dataset (1863) here. Try also showing the `id` column of the `history_dataset_association`.

> However, when deleting through the older UI, for histories that are identical in structure, I don't see the remaining files in the file system.

This is more confusing, but they should be using the same or similar API request for deleting or purging. Can you open your browser network console and monitor the request that gets sent to the server when deleting those datasets in both modes? That would point to a possible difference, otherwise, I can try to do it tomorrow.

**Comment by davelopez** [comment]
Ok, I think I managed to track down the difference between the old history and the new in terms of requesting the purge of a dataset.

The old history uses the old controller endpoint `purge_async` here:

https://github.com/galaxyproject/galaxy/blob/ca77e1841115991ee56c85dcf065d77dc23b1962/lib/galaxy/webapps/galaxy/controllers/dataset.py#L1012

This method ends up checking `hda.dataset.user_can_purge`:

https://github.com/galaxyproject/galaxy/blob/ca77e1841115991ee56c85dcf065d77dc23b1962/lib/galaxy/webapps/galaxy/controllers/dataset.py#L934

And, in this case, the result of the condition is `True` so it goes all the way and executes `hda.dataset.full_delete()` which effectively removes the file from the disk if no other associations are linked to the dataset. This explains why the datasets are instantly removed from the disk when using the old history.

On the other hand, the new history and the storage dashboard use the new `bulk_operation` and `delete_batch` API endpoints which checks similar conditions but with one difference. Since they process multiple HDAs in a loop, for performance reasons, the database flush is not done until the end of the loop. This causes the `hda.dataset.user_can_purge` condition to be `False` for each dataset because it expects the `hda.purged = True` to have been committed to the database beforehand.

This explains why only after the `cleanup_datasets` script is executed the datasets are effectively removed from disk.

Thank you @pcm32 for uncovering this bug!

**Comment by pcm32** [comment]
Glad that at least some of my issues are useful! since you guys have to put up with so many issues I create 😂 !

**Comment by pcm32** [comment]
Thanks for taking all this time for looking into it in detail @davelopez!

**Comment by davelopez** [comment]
I will assume this was hopefully fixed in https://github.com/galaxyproject/galaxy/pull/15305. If not, please feel free to reopen or create a new issue :)

**Comment by pcm32** [comment]
@davelopez I'm using a new container image that includes #15305 but I still the issue: purging deleted elements that are unique to a history doesn't get reflected in the file system, it is not actually deleting the files.

I took a history that had about 2 GBs of deleted files:

![image](https://user-images.githubusercontent.com/368478/227157388-6a50efc7-aff6-4906-87c5-cd143e522647.png)

at that point before deletion, storage looked like this:

![image](https://user-images.githubusercontent.com/368478/227157668-da22ce58-5271-4cdb-bc1b-85da82127c76.png)

I purged the deleted items from the little cog menu in the new history setup.

The history disk usage changed, but the the overall Galaxy count didn't, nor did the file system actual usage. Only when I went to the new interface to free up used space:

![image](https://user-images.githubusercontent.com/368478/227158169-476b23bb-39c5-420e-95cd-b08b01bf6e0c.png)

and deleted from there it reduced the used space of Galaxy:

![image](https://user-images.githubusercontent.com/368478/227158649-ee091bc0-eaff-4adc-8dd8-07e90e07853f.png)

however, the file system disk use still didn't change (used stayed at 310 GBs).

If I go and look into the specific purged items info:

![image](https://user-images.githubusercontent.com/368478/227159715-f79f6db5-4433-4efa-8ab4-23d161cd6a04.png)

the file is very much there in the file system:

```
galaxy@galaxy-prod-job-0-75f5cbdbdc-4bt47:/galaxy/server/objects$ ls -hl         /galaxy/server/objects/8/3/1/dataset_831c897a-d9db-4f43-828b-40983ca4f540.dat
-rw-r--r-- 1 galaxy galaxy 829M Mar 22 18:02 /galaxy/server/objects/8/3/1/dataset_831c897a-d9db-4f43-828b-40983ca4f540.dat
```

These are all intermediate files of a workflow that are left to be hidden (as they should be deleted), so they have been never added to another history or touched by other user. So on purging they should go away straightaway.

**Comment by pcm32** [comment]
In contrast, using the legacy history panel, a delete hidden items and then purge deleted items immediately gets rid of stuff on disk.

**Comment by mvdbeek** [comment]
I just tried it and it's working fine for me. You sure you have purging datasets enabled ? Are you using celery or not ? What's the state in the database, are the items purged ? Logs when you do a purge?

**Comment by pcm32** [comment]
- allow_user_dataset is unmodified (true by default AFAIK - also the legacy history purges items instantly from disk).
- yes, using celery
- I don't remember there being anything in particular in logs
- I would need to check the database, but on the UI the datasets have the purged icon.

**Comment by davelopez** [comment]
I will re-investigate this when I have a chance, but I'm surprised you are still seeing this behavior with #15305, we have now tests for that... https://github.com/galaxyproject/galaxy/pull/15305/files#diff-0feac75c1db5e9286380f23498ec73bea6c37c7ae0f5aab2c4a65ff2f22ff487

**Comment by davelopez** [comment]
Hey @pcm32, did you manage to confirm you had those commits in your instance?

I just tried this again, and it seems to work as expected:

Here I have a history after deleting (not purging) all its datasets:

![image](https://user-images.githubusercontent.com/46503462/235647751-c37d4604-a532-4a71-b365-ba35918193bb.png)

And this is the current state of the object store directory, where you can also see the shown `dataset_cfcf743b-bb2f-4c45-bb4e-06b78c5d35ea.dat` is still on disk:

![image](https://user-images.githubusercontent.com/46503462/235648108-1f507e96-b698-45d8-8128-76a43b326356.png)

I use `Purge all deleted content`:

![image](https://user-images.githubusercontent.com/46503462/235648425-dff6a0bf-844e-4263-8350-60534fef7109.png)

And I get:

![image](https://user-images.githubusercontent.com/46503462/235648690-b0b1de7c-00bd-40f1-b00d-a93e7ef7c96c.png)

![image](https://user-images.githubusercontent.com/46503462/235648859-41c7a9bf-2f3d-4d07-9cea-c1793898267c.png)

As you can see the space used is reduced and also the sample dataset is gone.

**Comment by pcm32** [comment]
The container I'm using is a branch that stems from release_22.05 at commit 5dc2c4f8c32656c25265f8ede086863ad11c5844 (dated 20th of March). If I look at the history at least, this includes your PR merged on the 16th of Jan. I will try it again and double check that the helm chart is replacing all the galaxy files with the ones from the container and that you changes are there.

**Comment by pcm32** [comment]
I have confirmed visually now as well that your changes are in my running instance.

**Comment by davelopez** [comment]
No worries! thank you so much for reporting back! :+1:
